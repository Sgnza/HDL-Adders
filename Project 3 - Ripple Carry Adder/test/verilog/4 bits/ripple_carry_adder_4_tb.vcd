$date
	Sat Jan 13 15:13:57 2024
$end
$version
	Icarus Verilog
$end
$timescale
	1ps
$end
$scope module ripple_carry_adder_4_tb $end
$var wire 4 ! sum_tb [3:0] $end
$var wire 1 " cout_tb $end
$var reg 4 # a_tb [3:0] $end
$var reg 4 $ b_tb [3:0] $end
$var reg 1 % cin_tb $end
$scope module rca0 $end
$var wire 4 & a [3:0] $end
$var wire 4 ' b [3:0] $end
$var wire 1 % cin $end
$var wire 1 " cout $end
$var wire 4 ( sum [3:0] $end
$var wire 1 ) fa3_sum $end
$var wire 1 * fa3_cout $end
$var wire 1 + fa2_sum $end
$var wire 1 , fa2_cout $end
$var wire 1 - fa1_sum $end
$var wire 1 . fa1_cout $end
$var wire 1 / fa0_sum $end
$var wire 1 0 fa0_cout $end
$scope module fa0 $end
$var wire 1 1 a $end
$var wire 1 2 b $end
$var wire 1 % cin $end
$var wire 1 0 cout $end
$var wire 1 / sum $end
$var wire 1 3 ha1_cout $end
$var wire 1 4 ha0_sum $end
$var wire 1 5 ha0_cout $end
$scope module ha0 $end
$var wire 1 1 a $end
$var wire 1 2 b $end
$var wire 1 4 sum $end
$var wire 1 5 cout $end
$upscope $end
$scope module hf1 $end
$var wire 1 % a $end
$var wire 1 4 b $end
$var wire 1 / sum $end
$var wire 1 3 cout $end
$upscope $end
$upscope $end
$scope module fa1 $end
$var wire 1 6 a $end
$var wire 1 7 b $end
$var wire 1 0 cin $end
$var wire 1 . cout $end
$var wire 1 - sum $end
$var wire 1 8 ha1_cout $end
$var wire 1 9 ha0_sum $end
$var wire 1 : ha0_cout $end
$scope module ha0 $end
$var wire 1 6 a $end
$var wire 1 7 b $end
$var wire 1 9 sum $end
$var wire 1 : cout $end
$upscope $end
$scope module hf1 $end
$var wire 1 0 a $end
$var wire 1 9 b $end
$var wire 1 - sum $end
$var wire 1 8 cout $end
$upscope $end
$upscope $end
$scope module fa2 $end
$var wire 1 ; a $end
$var wire 1 < b $end
$var wire 1 . cin $end
$var wire 1 , cout $end
$var wire 1 + sum $end
$var wire 1 = ha1_cout $end
$var wire 1 > ha0_sum $end
$var wire 1 ? ha0_cout $end
$scope module ha0 $end
$var wire 1 ; a $end
$var wire 1 < b $end
$var wire 1 > sum $end
$var wire 1 ? cout $end
$upscope $end
$scope module hf1 $end
$var wire 1 . a $end
$var wire 1 > b $end
$var wire 1 + sum $end
$var wire 1 = cout $end
$upscope $end
$upscope $end
$scope module fa3 $end
$var wire 1 @ a $end
$var wire 1 A b $end
$var wire 1 , cin $end
$var wire 1 * cout $end
$var wire 1 ) sum $end
$var wire 1 B ha1_cout $end
$var wire 1 C ha0_sum $end
$var wire 1 D ha0_cout $end
$scope module ha0 $end
$var wire 1 @ a $end
$var wire 1 A b $end
$var wire 1 C sum $end
$var wire 1 D cout $end
$upscope $end
$scope module hf1 $end
$var wire 1 , a $end
$var wire 1 C b $end
$var wire 1 ) sum $end
$var wire 1 B cout $end
$upscope $end
$upscope $end
$upscope $end
$scope task verify $end
$var reg 4 E a_task [3:0] $end
$var reg 4 F b_task [3:0] $end
$var reg 1 G cin_task $end
$var reg 1 H cout_task $end
$var reg 4 I sum_task [3:0] $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
$end
#0
$dumpvars
b0 I
1H
0G
b1 F
b1111 E
0D
1C
1B
0A
1@
0?
1>
1=
0<
1;
0:
19
18
07
16
15
04
03
12
11
10
0/
1.
0-
1,
0+
1*
0)
b0 (
b1 '
b1111 &
0%
b1 $
b1111 #
1"
b0 !
$end
#1
