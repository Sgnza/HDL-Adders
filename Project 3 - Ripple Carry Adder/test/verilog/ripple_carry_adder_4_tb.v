`timescale 1ps/1ps
`include "../../src/verilog/ripple_carry_adder_4.v"
module ripple_carry_adder_4_tb;

reg [3:0] a_tb;
reg [3:0] b_tb;
reg cin_tb;
wire [3:0] sum_tb;
wire cout_tb;

ripple_carry_adder_4 rca0(
    .a(a_tb),
    .b(b_tb),
    .cin(cin_tb),
    .sum(sum_tb),
    .cout(cout_tb)
);

task verify;
    input [3:0] a_task;
    input [3:0] b_task;
    input cin_task;
    input [3:0] sum_task;
    input cout_task;

    begin
        a_tb = a_task;
        b_tb = b_task;
        cin_tb = cin_task;
        #1;
        if(sum_task == sum_tb && cout_task == cout_tb) begin
            $display("CORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de entrada cin: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", a_tb, b_tb, cin_task, sum_task, sum_tb, cout_task, cout_tb);
        end else begin
            $display("INCORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de entrada cin: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", a_tb, b_tb, cin_task, sum_task, sum_tb, cout_task, cout_tb);

        end
    end

endtask

initial begin
  $dumpfile("ripple_carry_adder_4_tb.vcd");
  $dumpvars(0, ripple_carry_adder_4_tb);
end

initial begin

verify(
    4'b1111,
    4'b0001,
    1'b0,
    4'b0000,
    1'b1
);

end

endmodule