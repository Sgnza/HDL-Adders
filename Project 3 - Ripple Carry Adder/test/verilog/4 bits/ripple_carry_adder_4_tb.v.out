#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1539-g2693dd32b)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 12;
:vpi_module "C:\iverilog\lib\ivl\system.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "C:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "C:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "C:\iverilog\lib\ivl\va_math.vpi";
S_0000021354d4e1d0 .scope module, "ripple_carry_adder_4_tb" "ripple_carry_adder_4_tb" 2 3;
 .timescale -12 -12;
v0000021354da8e50_0 .var "a_tb", 3 0;
v0000021354da9c10_0 .var "b_tb", 3 0;
v0000021354da8f90_0 .var "cin_tb", 0 0;
v0000021354da90d0_0 .net "cout_tb", 0 0, L_0000021354d48170;  1 drivers
v0000021354da92b0_0 .net "sum_tb", 3 0, L_0000021354da9a30;  1 drivers
S_0000021354d3a320 .scope module, "rca0" "ripple_carry_adder_4" 2 11, 3 2 0, S_0000021354d4e1d0;
 .timescale -12 -12;
    .port_info 0 /INPUT 4 "a";
    .port_info 1 /INPUT 4 "b";
    .port_info 2 /INPUT 1 "cin";
    .port_info 3 /OUTPUT 1 "cout";
    .port_info 4 /OUTPUT 4 "sum";
L_0000021354d48170 .functor BUFZ 1, L_0000021354d48870, C4<0>, C4<0>, C4<0>;
v0000021354da89f0_0 .net "a", 3 0, v0000021354da8e50_0;  1 drivers
v0000021354da9d50_0 .net "b", 3 0, v0000021354da9c10_0;  1 drivers
v0000021354da8ef0_0 .net "cin", 0 0, v0000021354da8f90_0;  1 drivers
v0000021354da8c70_0 .net "cout", 0 0, L_0000021354d48170;  alias, 1 drivers
v0000021354da8bd0_0 .net "fa0_cout", 0 0, L_0000021354d48410;  1 drivers
v0000021354da9ad0_0 .net "fa0_sum", 0 0, L_0000021354d48020;  1 drivers
v0000021354da8770_0 .net "fa1_cout", 0 0, L_0000021354d48330;  1 drivers
v0000021354da9030_0 .net "fa1_sum", 0 0, L_0000021354d48aa0;  1 drivers
v0000021354da9850_0 .net "fa2_cout", 0 0, L_0000021354d48d40;  1 drivers
v0000021354da9170_0 .net "fa2_sum", 0 0, L_0000021354d48640;  1 drivers
v0000021354da9210_0 .net "fa3_cout", 0 0, L_0000021354d48870;  1 drivers
v0000021354da8db0_0 .net "fa3_sum", 0 0, L_0000021354d48db0;  1 drivers
v0000021354da8810_0 .net "sum", 3 0, L_0000021354da9a30;  alias, 1 drivers
L_0000021354da93f0 .part v0000021354da8e50_0, 0, 1;
L_0000021354da86d0 .part v0000021354da9c10_0, 0, 1;
L_0000021354da8950 .part v0000021354da8e50_0, 1, 1;
L_0000021354da8a90 .part v0000021354da9c10_0, 1, 1;
L_0000021354da9530 .part v0000021354da8e50_0, 2, 1;
L_0000021354da9cb0 .part v0000021354da9c10_0, 2, 1;
L_0000021354daa070 .part v0000021354da8e50_0, 3, 1;
L_0000021354da97b0 .part v0000021354da9c10_0, 3, 1;
L_0000021354da9a30 .concat [ 1 1 1 1], L_0000021354d48020, L_0000021354d48aa0, L_0000021354d48640, L_0000021354d48db0;
S_0000021354d3a4b0 .scope module, "fa0" "full_adder" 3 19, 4 2 0, S_0000021354d3a320;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /INPUT 1 "cin";
    .port_info 3 /OUTPUT 1 "sum";
    .port_info 4 /OUTPUT 1 "cout";
L_0000021354d48410 .functor OR 1, L_0000021354da9350, L_0000021354da9e90, C4<0>, C4<0>;
v0000021354d49160_0 .net "a", 0 0, L_0000021354da93f0;  1 drivers
v0000021354d49a20_0 .net "b", 0 0, L_0000021354da86d0;  1 drivers
v0000021354d49b60_0 .net "cin", 0 0, v0000021354da8f90_0;  alias, 1 drivers
v0000021354d49480_0 .net "cout", 0 0, L_0000021354d48410;  alias, 1 drivers
v0000021354d49700_0 .net "ha0_cout", 0 0, L_0000021354da9e90;  1 drivers
v0000021354d49de0_0 .net "ha0_sum", 0 0, L_0000021354d482c0;  1 drivers
v0000021354da6010_0 .net "ha1_cout", 0 0, L_0000021354da9350;  1 drivers
v0000021354da54d0_0 .net "sum", 0 0, L_0000021354d48020;  alias, 1 drivers
S_0000021354cecef0 .scope module, "ha0" "half_adder" 4 13, 5 1 0, S_0000021354d3a4b0;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000021354d482c0 .functor XOR 1, L_0000021354da93f0, L_0000021354da86d0, C4<0>, C4<0>;
v0000021354d49200_0 .net "a", 0 0, L_0000021354da93f0;  alias, 1 drivers
v0000021354d495c0_0 .net "b", 0 0, L_0000021354da86d0;  alias, 1 drivers
v0000021354d49ca0_0 .net "cout", 0 0, L_0000021354da9e90;  alias, 1 drivers
v0000021354d490c0_0 .net "sum", 0 0, L_0000021354d482c0;  alias, 1 drivers
L_0000021354da9e90 .arith/mult 1, L_0000021354da93f0, L_0000021354da86d0;
S_0000021354ced080 .scope module, "hf1" "half_adder" 4 22, 5 1 0, S_0000021354d3a4b0;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000021354d48020 .functor XOR 1, v0000021354da8f90_0, L_0000021354d482c0, C4<0>, C4<0>;
v0000021354d49660_0 .net "a", 0 0, v0000021354da8f90_0;  alias, 1 drivers
v0000021354d493e0_0 .net "b", 0 0, L_0000021354d482c0;  alias, 1 drivers
v0000021354d49340_0 .net "cout", 0 0, L_0000021354da9350;  alias, 1 drivers
v0000021354d49980_0 .net "sum", 0 0, L_0000021354d48020;  alias, 1 drivers
L_0000021354da9350 .arith/mult 1, v0000021354da8f90_0, L_0000021354d482c0;
S_0000021354d02590 .scope module, "fa1" "full_adder" 3 27, 4 2 0, S_0000021354d3a320;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /INPUT 1 "cin";
    .port_info 3 /OUTPUT 1 "sum";
    .port_info 4 /OUTPUT 1 "cout";
L_0000021354d48330 .functor OR 1, L_0000021354da88b0, L_0000021354da9490, C4<0>, C4<0>;
v0000021354da48f0_0 .net "a", 0 0, L_0000021354da8950;  1 drivers
v0000021354da4ad0_0 .net "b", 0 0, L_0000021354da8a90;  1 drivers
v0000021354da59d0_0 .net "cin", 0 0, L_0000021354d48410;  alias, 1 drivers
v0000021354da5f70_0 .net "cout", 0 0, L_0000021354d48330;  alias, 1 drivers
v0000021354da61f0_0 .net "ha0_cout", 0 0, L_0000021354da9490;  1 drivers
v0000021354da66f0_0 .net "ha0_sum", 0 0, L_0000021354d48100;  1 drivers
v0000021354da6790_0 .net "ha1_cout", 0 0, L_0000021354da88b0;  1 drivers
v0000021354da5d90_0 .net "sum", 0 0, L_0000021354d48aa0;  alias, 1 drivers
S_0000021354d02720 .scope module, "ha0" "half_adder" 4 13, 5 1 0, S_0000021354d02590;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000021354d48100 .functor XOR 1, L_0000021354da8950, L_0000021354da8a90, C4<0>, C4<0>;
v0000021354da5110_0 .net "a", 0 0, L_0000021354da8950;  alias, 1 drivers
v0000021354da5a70_0 .net "b", 0 0, L_0000021354da8a90;  alias, 1 drivers
v0000021354da56b0_0 .net "cout", 0 0, L_0000021354da9490;  alias, 1 drivers
v0000021354da52f0_0 .net "sum", 0 0, L_0000021354d48100;  alias, 1 drivers
L_0000021354da9490 .arith/mult 1, L_0000021354da8950, L_0000021354da8a90;
S_0000021354d028b0 .scope module, "hf1" "half_adder" 4 22, 5 1 0, S_0000021354d02590;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000021354d48aa0 .functor XOR 1, L_0000021354d48410, L_0000021354d48100, C4<0>, C4<0>;
v0000021354da5bb0_0 .net "a", 0 0, L_0000021354d48410;  alias, 1 drivers
v0000021354da5e30_0 .net "b", 0 0, L_0000021354d48100;  alias, 1 drivers
v0000021354da4f30_0 .net "cout", 0 0, L_0000021354da88b0;  alias, 1 drivers
v0000021354da5570_0 .net "sum", 0 0, L_0000021354d48aa0;  alias, 1 drivers
L_0000021354da88b0 .arith/mult 1, L_0000021354d48410, L_0000021354d48100;
S_0000021354da68b0 .scope module, "fa2" "full_adder" 3 35, 4 2 0, S_0000021354d3a320;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /INPUT 1 "cin";
    .port_info 3 /OUTPUT 1 "sum";
    .port_info 4 /OUTPUT 1 "cout";
L_0000021354d48d40 .functor OR 1, L_0000021354da9990, L_0000021354da98f0, C4<0>, C4<0>;
v0000021354da4a30_0 .net "a", 0 0, L_0000021354da9530;  1 drivers
v0000021354da6290_0 .net "b", 0 0, L_0000021354da9cb0;  1 drivers
v0000021354da4990_0 .net "cin", 0 0, L_0000021354d48330;  alias, 1 drivers
v0000021354da57f0_0 .net "cout", 0 0, L_0000021354d48d40;  alias, 1 drivers
v0000021354da5430_0 .net "ha0_cout", 0 0, L_0000021354da98f0;  1 drivers
v0000021354da5890_0 .net "ha0_sum", 0 0, L_0000021354d48790;  1 drivers
v0000021354da5cf0_0 .net "ha1_cout", 0 0, L_0000021354da9990;  1 drivers
v0000021354da63d0_0 .net "sum", 0 0, L_0000021354d48640;  alias, 1 drivers
S_0000021354da6a40 .scope module, "ha0" "half_adder" 4 13, 5 1 0, S_0000021354da68b0;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000021354d48790 .functor XOR 1, L_0000021354da9530, L_0000021354da9cb0, C4<0>, C4<0>;
v0000021354da4c10_0 .net "a", 0 0, L_0000021354da9530;  alias, 1 drivers
v0000021354da60b0_0 .net "b", 0 0, L_0000021354da9cb0;  alias, 1 drivers
v0000021354da5750_0 .net "cout", 0 0, L_0000021354da98f0;  alias, 1 drivers
v0000021354da6150_0 .net "sum", 0 0, L_0000021354d48790;  alias, 1 drivers
L_0000021354da98f0 .arith/mult 1, L_0000021354da9530, L_0000021354da9cb0;
S_0000021354da6bd0 .scope module, "hf1" "half_adder" 4 22, 5 1 0, S_0000021354da68b0;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000021354d48640 .functor XOR 1, L_0000021354d48330, L_0000021354d48790, C4<0>, C4<0>;
v0000021354da6330_0 .net "a", 0 0, L_0000021354d48330;  alias, 1 drivers
v0000021354da5390_0 .net "b", 0 0, L_0000021354d48790;  alias, 1 drivers
v0000021354da5c50_0 .net "cout", 0 0, L_0000021354da9990;  alias, 1 drivers
v0000021354da4b70_0 .net "sum", 0 0, L_0000021354d48640;  alias, 1 drivers
L_0000021354da9990 .arith/mult 1, L_0000021354d48330, L_0000021354d48790;
S_0000021354da7d70 .scope module, "fa3" "full_adder" 3 43, 4 2 0, S_0000021354d3a320;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /INPUT 1 "cin";
    .port_info 3 /OUTPUT 1 "sum";
    .port_info 4 /OUTPUT 1 "cout";
L_0000021354d48870 .functor OR 1, L_0000021354da9670, L_0000021354da95d0, C4<0>, C4<0>;
v0000021354da6650_0 .net "a", 0 0, L_0000021354daa070;  1 drivers
v0000021354da4d50_0 .net "b", 0 0, L_0000021354da97b0;  1 drivers
v0000021354da4df0_0 .net "cin", 0 0, L_0000021354d48d40;  alias, 1 drivers
v0000021354da4e90_0 .net "cout", 0 0, L_0000021354d48870;  alias, 1 drivers
v0000021354da4fd0_0 .net "ha0_cout", 0 0, L_0000021354da95d0;  1 drivers
v0000021354da5250_0 .net "ha0_sum", 0 0, L_0000021354d48800;  1 drivers
v0000021354da5070_0 .net "ha1_cout", 0 0, L_0000021354da9670;  1 drivers
v0000021354da51b0_0 .net "sum", 0 0, L_0000021354d48db0;  alias, 1 drivers
S_0000021354da7f00 .scope module, "ha0" "half_adder" 4 13, 5 1 0, S_0000021354da7d70;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000021354d48800 .functor XOR 1, L_0000021354daa070, L_0000021354da97b0, C4<0>, C4<0>;
v0000021354da6510_0 .net "a", 0 0, L_0000021354daa070;  alias, 1 drivers
v0000021354da5930_0 .net "b", 0 0, L_0000021354da97b0;  alias, 1 drivers
v0000021354da5ed0_0 .net "cout", 0 0, L_0000021354da95d0;  alias, 1 drivers
v0000021354da4cb0_0 .net "sum", 0 0, L_0000021354d48800;  alias, 1 drivers
L_0000021354da95d0 .arith/mult 1, L_0000021354daa070, L_0000021354da97b0;
S_0000021354da8090 .scope module, "hf1" "half_adder" 4 22, 5 1 0, S_0000021354da7d70;
 .timescale -12 -12;
    .port_info 0 /INPUT 1 "a";
    .port_info 1 /INPUT 1 "b";
    .port_info 2 /OUTPUT 1 "sum";
    .port_info 3 /OUTPUT 1 "cout";
L_0000021354d48db0 .functor XOR 1, L_0000021354d48d40, L_0000021354d48800, C4<0>, C4<0>;
v0000021354da6470_0 .net "a", 0 0, L_0000021354d48d40;  alias, 1 drivers
v0000021354da5b10_0 .net "b", 0 0, L_0000021354d48800;  alias, 1 drivers
v0000021354da5610_0 .net "cout", 0 0, L_0000021354da9670;  alias, 1 drivers
v0000021354da65b0_0 .net "sum", 0 0, L_0000021354d48db0;  alias, 1 drivers
L_0000021354da9670 .arith/mult 1, L_0000021354d48d40, L_0000021354d48800;
S_0000021354daa230 .scope task, "verify" "verify" 2 19, 2 19 0, S_0000021354d4e1d0;
 .timescale -12 -12;
v0000021354da8d10_0 .var "a_task", 3 0;
v0000021354da8450_0 .var "b_task", 3 0;
v0000021354da8b30_0 .var "cin_task", 0 0;
v0000021354da9b70_0 .var "cout_task", 0 0;
v0000021354da9710_0 .var "sum_task", 3 0;
TD_ripple_carry_adder_4_tb.verify ;
    %load/vec4 v0000021354da8d10_0;
    %store/vec4 v0000021354da8e50_0, 0, 4;
    %load/vec4 v0000021354da8450_0;
    %store/vec4 v0000021354da9c10_0, 0, 4;
    %load/vec4 v0000021354da8b30_0;
    %store/vec4 v0000021354da8f90_0, 0, 1;
    %delay 1, 0;
    %load/vec4 v0000021354da9710_0;
    %load/vec4 v0000021354da92b0_0;
    %cmp/e;
    %flag_get/vec4 4;
    %jmp/0 T_0.2, 4;
    %load/vec4 v0000021354da9b70_0;
    %load/vec4 v0000021354da90d0_0;
    %cmp/e;
    %flag_get/vec4 4;
    %and;
T_0.2;
    %flag_set/vec4 8;
    %jmp/0xz  T_0.0, 8;
    %vpi_call 2 32 "$display", "CORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de entrada cin: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", v0000021354da8e50_0, v0000021354da9c10_0, v0000021354da8b30_0, v0000021354da9710_0, v0000021354da92b0_0, v0000021354da9b70_0, v0000021354da90d0_0 {0 0 0};
    %jmp T_0.1;
T_0.0 ;
    %vpi_call 2 34 "$display", "INCORRECTO: Dato de entrada a: %b, Dato de entrada b: %b, Dato de entrada cin: %b, Dato de salida s esperado: %b, Dato de salida s obtenido: %b, Dato de salida c esperado: %b, Dato de salida c obtenido: %b", v0000021354da8e50_0, v0000021354da9c10_0, v0000021354da8b30_0, v0000021354da9710_0, v0000021354da92b0_0, v0000021354da9b70_0, v0000021354da90d0_0 {0 0 0};
T_0.1 ;
    %end;
    .scope S_0000021354d4e1d0;
T_1 ;
    %vpi_call 2 42 "$dumpfile", "ripple_carry_adder_4_tb.vcd" {0 0 0};
    %vpi_call 2 43 "$dumpvars", 32'sb00000000000000000000000000000000, S_0000021354d4e1d0 {0 0 0};
    %end;
    .thread T_1;
    .scope S_0000021354d4e1d0;
T_2 ;
    %pushi/vec4 15, 0, 4;
    %store/vec4 v0000021354da8d10_0, 0, 4;
    %pushi/vec4 1, 0, 4;
    %store/vec4 v0000021354da8450_0, 0, 4;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v0000021354da8b30_0, 0, 1;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0000021354da9710_0, 0, 4;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v0000021354da9b70_0, 0, 1;
    %fork TD_ripple_carry_adder_4_tb.verify, S_0000021354daa230;
    %join;
    %end;
    .thread T_2;
# The file index is used to find the file name in the following table.
:file_names 6;
    "N/A";
    "<interactive>";
    "ripple_carry_adder_4_tb.v";
    "./../../src/verilog/ripple_carry_adder_4.v";
    "./../../../Project 2 - Full Adder/src/verilog/full_adder.v";
    "./../../../Project 1 - Half Adder/src/verilog/half_adder.v";
